File: in-motion-server/Friends/IMS.Friends.Models/Dto/Outgoing/AcceptedFriendshipDto.cs
namespace IMS.Friends.Models.Dto.Outgoing;

public class AcceptedFriendshipDto
{
    public string Id { get; set; }
    
    public string ExternalUserId { get; set; }
    
    public FriendInfoDto ExternalUser { get; set; }
    
    public DateTime FriendsSince { get; set; }
}

File: in-motion-android/app/src/main/java/com/inmotion/in_motion_android/data/remote/dto/friends/AcceptedFriendshipDto.kt
package com.inmotion.in_motion_android.data.remote.dto.friends

import com.inmotion.in_motion_android.data.database.entity.AcceptedFriend

data class AcceptedFriendshipDto(
    val id: String,
    val externalUserId: String,
    val externalUser: FriendInfoDto,
    val friendsSince: String
) {
    fun toAcceptedFriend(): AcceptedFriend {
        return AcceptedFriend(
            externalUserId,
            id,
            externalUser.nickname,
            externalUser.bio,
            friendsSince
        )
    }
}
